cpp/src/join/joining.cu

```diff
@@ -399,83 +393,121 @@ gdf_error construct_join_output_df(
-                                   (nullptr != right_indices)};
+  std::vector <gdf_column*> tmp_left_cols;
+  std::vector <gdf_column*> tmp_right_cols;
 
-  GDF_REQUIRE(construct_output_dataframe or return_output_indices,
-              GDF_INVALID_API_CALL);
+  for (int i=0; i < num_left_cols; ++i)
+  {
+      tmp_left_cols.push_back(const_cast<gdf_column *> (left_cols.get_column(i)));
+  }
+
+  for (int i=0; i < num_right_cols; ++i)
+  {
+      if (std::find(r_common_name_join_ind.begin(), r_common_name_join_ind.end(), i)
+            == r_common_name_join_ind.end()) {
+          tmp_right_cols.push_back(const_cast<gdf_column *> (right_cols.get_column(i)));
+      }
+  }
 
-  auto const left_col_size = left_cols[0]->size;
-  auto const right_col_size = right_cols[0]->size;
+  cudf::table tmp_left_table = (tmp_left_cols.size()>0)? cudf::table (tmp_left_cols) : cudf::table{};
+  cudf::table tmp_right_table = (tmp_right_cols.size()>0)? cudf::table (tmp_right_cols) : cudf::table{};
+
+  CUDF_EXPECTS (0 != num_left_cols, "Left table is empty");
+  CUDF_EXPECTS (0 != num_right_cols, "Right table is empty");
+  CUDF_EXPECTS (nullptr != join_context, "Join context is invalid");
+
+  if (0 == num_cols_to_join)
+  {
+      return std::pair <cudf::table, cudf::table>(cudf::empty_like(tmp_left_table), cudf::empty_like(tmp_right_table));
+  }
+
+  for(int column_index = 0; column_index  < num_left_cols; column_index++){
+    CUDF_EXPECTS(left_cols.get_column(column_index)->dtype != GDF_invalid, "Unsupported Dtype in Left column");
+  }
+  for(int column_index = 0; column_index  < num_right_cols; column_index++){
+    CUDF_EXPECTS(right_cols.get_column(column_index)->dtype != GDF_invalid, "Unsupported Dtype in right column");
+  }
+
+  auto const left_col_size = left_cols.get_column(0)->size;
+  auto const right_col_size = right_cols.get_column(0)->size;
 
   // If the inputs are empty, immediately return
   if ((0 == left_col_size) && (0 == right_col_size)) {
-    return GDF_SUCCESS;
+      return std::pair <cudf::table, cudf::table>(cudf::empty_like(tmp_left_table), cudf::empty_like(tmp_right_table));
   }
 
   // If left join and the left table is empty, return immediately
   if ((JoinType::LEFT_JOIN == join_type) && (0 == left_col_size)) {
-    return GDF_SUCCESS;
+      return std::pair <cudf::table, cudf::table>(cudf::empty_like(tmp_left_table), cudf::empty_like(tmp_right_table));
   }
 
   // If Inner Join and either table is empty, return immediately
   if ((JoinType::INNER_JOIN == join_type) &&
       ((0 == left_col_size) || (0 == right_col_size))) {
-    return GDF_SUCCESS;
+      return std::pair <cudf::table, cudf::table>(cudf::empty_like(tmp_left_table), cudf::empty_like(tmp_right_table));
   }
 
 
   //if the inputs are nvcategory we need to make the dictionaries comparable
   bool at_least_one_category_column = false;
   for(int join_column_index = 0; join_column_index < num_cols_to_join; join_column_index++){
```

Comment:
Replace this with a one-liner using `bool at_least_one_category_column == std::any_of(...)`.